AlphaWallet/Transactions/Storage/TransactionsStorage.swift

```diff
@@ -28,6 +28,10 @@ class TransactionsStorage {
         return objects.filter { $0.state == .completed }
     }
 
+    var transactionObjectsThatDoNotComeFromEventLogs: [Transaction] {
+        return objects.filter { !$0.isERC20Interaction }
```

Comment:
Can you change this to (untested code):

```
        return objects
            .filter("state == \(TransactionState.completed.rawValue)")
            .filter("isERC20Interaction == false")
```

The non-string version is ran as normal Swift collection, so the filter() call forces the entire transactions dataset to be loaded in memory and every Transaction instance runs the block against it.

The string version is run natively within Realm without having to load the entire transactions dataset.

(I know the `completedObjects` still use the block version, we inherited them and we haven't fix it yet)

Reply:
@James-Sangalli Updated the property to:

```
    var transactionObjectsThatDoNotComeFromEventLogs: Results<Transaction> {
        return realm.objects(Transaction.self)
                .sorted(byKeyPath: "date", ascending: false)
                .filter("chainId = \(self.server.chainID)")
                .filter("id != ''")
                .filter("state == \(TransactionState.completed.rawValue)")
                .filter("isERC20Interaction == false")
    }
```

Some duplicate in the filtering, but I didn't want to make you have to make additional changes to the existing `object` property in the PR.